lib/text/preprocessor.js

```diff
@@ -4,13 +4,15 @@ class Preprocessor extends SHRParserVisitor {
   constructor() {
     super();
     this._data = new PreprocessedData();
+    this._vsData = new PreprocessedData();
   }
 
   get data() { return this._data; }
+  get vsData() { return this._vsData; }
 
   visitShr(ctx) {
     if (ctx.dataDefsDoc()) {
-      return this.visitDataDefsDoc(ctx.dataDefsDoc());
+        return this.visitDataDefsDoc(ctx.dataDefsDoc());
     } else {
```

Comment:
I'd change the `else` to `else if (ctx.valuesetDefsDoc())`.  There will soon be another possibility here (for mapping docs), so it's best to be careful.